haystack/components/converters/jq_json.py

```diff
@@ -0,0 +1,133 @@
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union, Iterator, Callable
+
+from haystack import Document, component, logging
+from haystack.components.converters.utils import get_bytestream_from_source, normalize_metadata
+from haystack.lazy_imports import LazyImport
+
+import json
+
+with LazyImport("Run 'pip install jq' Github - https://github.com/jqlang/jq ") as jq_import:
+    import jq
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class JQToJSON:
```

Comment:
> @CuriousLearner2 - Would name this `JSONToDocument` to be consistent with other converters in the Haystack pipeline.

Here are the converters.  Seems like it could fit.  Did you have any other suggestions? https://docs.haystack.deepset.ai/docs/converters

Reply:
Hmmm. It is strange that they don't all follow the exact same naming convention. I think the convention `____ToDocument` makes the most sense because we are converting a file of type ____ to a haystack `Document` object.

Reply:
So no change, correct?

Reply:
Yes. No change for now.

Reply:
How about JQ_JSONToDocument.  I would like to have JQ in there.

Reply:
`JSONToDocument` seems more accurate to me.

Reply:
Doesn't this name suggest that we are going straight from JSON to a Document without any post-processing by jq?  I would prefer to see a reference to jq since there are other choices we can make about how to convert JSON to a Document.

Reply:
But you could see that other converters like `DocxToDocument` for eg. use `python-docx` under the hood. But this is not visible. Hence I feel eventhough we use `jq` that doesn't need to be exposed or known to the user.

Reply:
One distinction between DocxToDocument and this converter is that jq is a parameter. 'jq' is not 'under the hood'. 
What would we call a converter that converts json to a Doc without using jq?  Would they both be called JSONToDocument?

Reply:
As a point of note we see that Langchain calls their class JSONLoader which also uses jq as a parameter.  They do not reference jq in the name of the class, so possibly JSONToDocument might be sufficient. https://github.com/langchain-ai/langchain/blob/9be6b5a20ffa996a15f537ccf071fc9da6e344a0/libs/community/langchain_community/document_loaders/json_loader.py#L164

Reply:
After mulling it over I agree with the two of you on the naming convention.  Let's call the class JSONToDocument